colossalai/shardformer/shard/shard_config.py

```diff
@@ -109,8 +109,23 @@ def _turn_on_all_optimization(self):
         Turn on all optimization.
         """
         # you can add all the optimization flag here
-        self.enable_fused_normalization = True
-        self.enable_flash_attention = True
+        try:
+            from apex.normalization import FusedLayerNorm as ApexFusedLayerNorm  # noqa
+
+            apex_installed = True
+        except ImportError:
+            apex_installed = False
+            warnings.warn("You set enable_all_optimization=True, but apex is not installed.")
+        try:
+            import flash_attn  # noqa
```

Comment:
flash_attn is not essential for `enable_flash_attention=True`. if flash_attn is not installed, it will use torch sdpa

Reply:
So this will not occur if torch 2.2 is installed?
<img width="872" alt="image" src="https://github.com/hpcaitech/ColossalAI/assets/87317405/cdd0d245-dd56-468a-ad0e-093681fa1cda">